components/clientComponents/forms/AddressComplete/AddressComplete.tsx

```diff
@@ -0,0 +1,309 @@
+"use client";
+import { AddressCompleteChoice, AddressCompleteProps, AddressElements } from "./types";
+import { getAddressCompleteChoices, getSelectedAddress } from "./utils";
+import { Combobox, Description, Label, ManagedCombobox } from "@clientComponents/forms";
+import { useState, useEffect, useRef } from "react";
+import { useTranslation } from "@i18n/client";
+import { useField } from "formik";
+import { cn } from "@lib/utils";
+import { SpinnerIcon } from "@serverComponents/icons/SpinnerIcon";
+import { Language } from "@lib/types/form-builder-types";
+import { countries } from "@lib/managedData/countries";
+
+interface ManagedComboboxRef {
+  changeInputValue: (value: string) => void;
+}
+
+export const AddressComplete = (props: AddressCompleteProps): React.ReactElement => {
+  const { id, name, required, ariaDescribedBy } = props;
+
+  const [field, meta, helpers] = useField(props);
+
+  const { t, i18n } = useTranslation("form-builder");
+
+  //Address Complete elements
+  const [choices, setChoices] = useState<string[]>([]);
+  const [addressResultCache, setAddressResultCache] = useState<AddressCompleteChoice[]>([]); // Cache the results from the address search.
+
+  // Cache and Allow values
+  const [isReady, setIsReady] = useState(false);
+  const [allow, setAllow] = useState(false);
+  const [apiKey, setApiKey] = useState("");
+
+  const toFullAddress = (address: AddressCompleteChoice): string => {
+    return address.Text + ", " + address.Description;
+  };
+
+  const comboboxRef = useRef<ManagedComboboxRef>(null);
+
+  // Check if addressComplete is allowed.
+  useEffect(() => {
+    const checkAllowed = async () => {
+      const response = await fetch("/api/components", { method: "POST" });
+      const jsonData = await response.json();
+      setAllow(jsonData.allowed);
+      setApiKey(jsonData.key);
+      setIsReady(true);
+    };
+
+    if (apiKey === "") {
+      // Don't recheck.
+      checkAllowed();
+    }
+  }, [apiKey]);
```

Comment:
Shouldn't this just be controlled by the feature flag?
Is there a better way of providing the API key? It's not sensitive, could the production and staging keys just be embedded and switched based on APP_ENV?

Reply:
This was before the fun feature flag provider was made, will refactor.